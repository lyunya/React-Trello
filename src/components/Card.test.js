import React from 'react';
import ReactDOM from 'react-dom';
import Card from './Card';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
    // first create a DOM element to render the component into
    const div = document.createElement('div');
  
    // render the component, this is the actual test, if something is wrong it will fail here
    ReactDOM.render(<Card key = {0} title={'title'} content={'item.content'}/>, div);
  
    // clean up code
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the Card UI as expected', () => {
    const tree = renderer
      .create(<Card key = {0} title={'title'} content={'item.content'}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
    });