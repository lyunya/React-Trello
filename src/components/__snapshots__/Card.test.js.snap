// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Card UI as expected 1`] = `
<div
  className="Card"
>
  <button
    type="button"
  >
    delete
  </button>
  <h3>
    title
  </h3>
  <p>
    item.content
  </p>
</div>
`;
